src/main/java/example/docvalues/DocValuesSearchExample.java

```diff
@@ -0,0 +1,140 @@
+// # Doc values example
+//
+// This example will cover the basics of the Apache Lucene doc values, including storing, querying and retrieving
+// them (for the search hits).
+//
+// 1. https://www.elastic.co/guide/en/elasticsearch/reference/7.10/doc-values.html#doc-values
+// 2. https://www.elastic.co/blog/sparse-versus-dense-document-values-with-apache-lucene
```

Comment:
Other useful references:

* https://trifork.nl/blog/introducing-lucene-index-doc-values/
* https://j.blaszyk.me/tech-blog/exploring-apache-lucene-index/#docvalues
* https://www.youtube.com/watch?v=JLUgEpJcG40

Reply:
I will add them shortly, thanks @msfroh !

Reply:
Haha .. you merge it before :D